File: app/src/main/java/com/ywwynm/everythingdone/adapters/BaseViewHolder.java
package com.ywwynm.everythingdone.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by ywwynm on 2016/6/23.
 * basic ViewHolder which provides f() for a simpler way to findViewById
 */
public class BaseViewHolder extends RecyclerView.ViewHolder {

    protected View mContentView;

    public BaseViewHolder(View itemView) {
        super(itemView);
        mContentView = itemView;
    }

    @SuppressWarnings("unchecked")
    protected final <T extends View> T f(int id) {
        return (T) mContentView.findViewById(id);
    }
}

File: app/src/main/java/com/ywwynm/everythingdone/adapters/BaseViewHolder.kt
package com.ywwynm.everythingdone.adapters

import androidx.recyclerview.widget.RecyclerView
import android.view.View

/**
 * Created by ywwynm on 2016/6/23.
 * Translated to Kotlin by ywwynm and Claude Opus 4.7 on 2026/5/20.
 * basic ViewHolder which provides f() for a simpler way to findViewById
 */
open class BaseViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {

    protected var mContentView: View? = itemView

    @Suppress("UNCHECKED_CAST")
    protected fun <T : View?> f(id: Int): T {
        return mContentView!!.findViewById<View>(id) as T
    }
}
